python/az/aro/azext_aro/_aad.py

```diff
@@ -71,3 +78,35 @@ def create_service_principal(self, app_id):
                 retries += 1
                 logger.warning("%s; retry %d of %d", ex, retries, max_retries)
                 time.sleep(10)
+
+    def refresh_application_credentials(self, object_id):
+        password = uuid.uuid4()
+        key_id = uuid.uuid4()
+        start_date = datetime.datetime.utcnow()
+        end_date = datetime.datetime(2299, 12, 31, tzinfo=datetime.timezone.utc)
+
+        try:
+            credentials = list(self.client.applications.list_password_credentials(object_id))
+        except GraphErrorException as e:
+            logger.error(e.message)
+            raise
```

Comment:
@mjudeikis looking back at the previous PR we merged into azure-cli, I realized that they changed their [error guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#Error-Type).  I'm thinking we might have to update any errors we're raising that aren't in the lowest level list. 

Wondering if they might hit us on this when merging into upstream.  Might want to do the following in all cases with a`BadRequestError` from `azure.cli.core.azclierror`.  

```suggestion
            raise BadRequestError(e.message)
```

Reply:
There's a few places where it would need to be update.  Potentially in a follow-up PR we could update all of them and not just the ones touched.

Reply:
Gonna do this as follow-up. This is already :exploding_head: